components/user/UserProfile.js

```diff
@@ -68,34 +69,77 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       </div>
       <div className="flex justify-center my-4 text-center">
-        <ReactMarkdown components={{ a: LinkRenderer }}>{data.bio}</ReactMarkdown>
+        <ReactMarkdown components={{ a: LinkRenderer }}>
+          {data.bio}
+        </ReactMarkdown>
       </div>
       {!qrShow && (
         <div className="flex flex-wrap justify-center">
           {data.tags &&
             data.tags.map((tag) => (
-              <Tag name={tag} key={tag.toLowerCase()} onClick={() => router.push(`/search?keyword=${tag.toLowerCase()}`)} />
+              <Tag
+                name={tag}
+                key={tag.toLowerCase()}
+                onClick={() =>
+                  router.push(`/search?keyword=${tag.toLowerCase()}`)
+                }
+              />
             ))}
         </div>
       )}
 
       {/* Passed Ref object as the ref attribute to the JSX of the DOM node of QR */}
-      <div className="flex justify-center my-4" ref={qrRef}>
-        {qrShow && (
-          <QRCodeCanvas
-            className="border border-white"
-            value={`${BASE_URL}/${data.username}`}
-            size={fallbackImageSize * 2}
-          />
-        )}
-      </div>
-      <div className="flex justify-center mb-4">
-        {qrShow && (
-          <Button primary={true} onClick={downloadQR}>
-            Download QR code
-          </Button>
-        )}
-      </div>
+      <Modal show={qrShow} setShow={setQrShow} modalStyles="w-fit m-auto">
+        <div className="flex flex-col items-center justify-center px-8">
+          <div>
+            <div className="flex justify-center my-4" ref={qrRef}>
+              {qrShow && (
+                <QRCodeCanvas
+                  className="border border-white"
+                  value={`${BASE_URL}/${data.username}`}
+                  size={fallbackImageSize * 2}
+                />
+              )}
+            </div>
+            <div className="w-[100%] px-2 mx-auto flex justify-center mb-4">
+              {qrShow && (
+                <Button primary={true} onClick={downloadQR}>
+                  Download QR code
+                </Button>
+              )}
+            </div>
+          </div>
+          {qrShow && (
+            <>
+              <div className="h-full m-4 p-2 flex flex-row items-start justify-center space-x-2">
+                {socials.map(({ href, Icon, includeText }) => (
+                  <a
+                    key={href}
+                    href={`${href}${BASE_URL}/${data.username}${
```

Comment:
Not sure this looks right? `BASE_URL` should be at the beginning

Reply:
I am passing `BASE_URL` as a url param int the href url to share the profile through the chosen social media channel. You can find the href in the config/socials.js in the socials array

Reply:
Can you share an example of the final value string? Or maybe you have a different value for `BASE_URL` from which we use in the app - which is confusing me 🤔 if it is a different value, maybe use a different name so others don't get confused, like `SOCIAL_URL`

Reply:
`https://twitter.com/intent/tweet?url=https://linkfree.io/shoebilyas123`
This is how the href link will look. I agree, that it would be better to change it to SOCIAL_URL or SOCIAL_SHARE_LINK. It won't anyone confuse this way.